File: src/main/java/net/dontdrinkandroot/gitki/wicket/choicerenderer/IsoLanguageChoiceRenderer.java
package net.dontdrinkandroot.gitki.wicket.choicerenderer;

import org.apache.wicket.markup.html.form.IChoiceRenderer;
import org.apache.wicket.model.IModel;

import java.util.List;
import java.util.Locale;

/**
 * @author Philip Washington Sorst <philip@sorst.net>
 */
public class IsoLanguageChoiceRenderer implements IChoiceRenderer<String>
{
    @Override
    public Object getDisplayValue(String object)
    {
        Locale locale = new Locale(object);
        return locale.getDisplayLanguage(locale);
    }

    @Override
    public String getIdValue(String object, int index)
    {
        return object;
    }

    @Override
    public String getObject(String id, IModel<? extends List<? extends String>> choices)
    {
        return id;
    }
}

File: src/main/kotlin/net/dontdrinkandroot/gitki/wicket/choicerenderer/IsoLanguageChoiceRenderer.kt
package net.dontdrinkandroot.gitki.wicket.choicerenderer

import org.apache.wicket.markup.html.form.IChoiceRenderer
import org.apache.wicket.model.IModel
import java.util.*

class IsoLanguageChoiceRenderer : IChoiceRenderer<String> {

    override fun getDisplayValue(language: String): Any = Locale(language).let { it.getDisplayLanguage(it) }

    override fun getIdValue(language: String, index: Int): String = language

    override fun getObject(id: String, choices: IModel<out MutableList<out String>>): String = id
}